src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureSelector.h

```diff
@@ -0,0 +1,265 @@
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Douglas McCloskey, Pasquale Domenico Colaianni, Svetlana Kutuzova $
+// $Authors: Douglas McCloskey, Pasquale Domenico Colaianni, Svetlana Kutuzova $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/config.h> // OPENMS_DLLAPI
+#include <OpenMS/DATASTRUCTURES/LPWrapper.h>
+#include <OpenMS/DATASTRUCTURES/String.h>
+#include <OpenMS/KERNEL/FeatureMap.h>
+
+namespace OpenMS
+{
+  class OPENMS_DLLAPI MRMFeatureSelector
+  {
+public:
+    MRMFeatureSelector() = default;
+    virtual ~MRMFeatureSelector() = default;
+
+    /// To test private and protected methods
+    friend class MRMFeatureSelector_test;
+
+    /**
+      Derived classes implement this pure virtual method.
+
+      It setups the linear programming problem and solves it.
+
+      @param[in] time_to_name Pairs representing a mapping of retention times to transition names
+      @param[in] feature_name_map Transitions' names to their features objects
+      @param[out] result Transitions' names filtered out of the LP problem
+    */
+    virtual void optimize(
+      const std::vector<std::pair<double, String>>& time_to_name,
+      const std::map<String, std::vector<Feature>>& feature_name_map,
+      std::vector<String>& result
+    ) = 0;
+
+    /**
+      Splits the features into time segments. `optimize()` method is run on each of these segments.
```

Comment:
if possible some additional information should be added to the comment. e.g., are there any constraints on the segments? are these non-overlapping? etc.